src/iperf_api.c

```diff
@@ -2728,8 +2728,9 @@ JSON_read(int fd)
      * Then read the JSON into a buffer and parse it.  Return a parsed JSON
      * structure, NULL if there was an error.
      */
-    if (Nread(fd, (char*) &nsize, sizeof(nsize), Ptcp) >= 0) {
-	hsize = ntohl(nsize);
+    rc = Nread(fd, (char*) &nsize, sizeof(nsize), Ptcp);
+    hsize = ntohl(nsize);
+    if (rc == sizeof(nsize) && hsize > 0 && hsize <= MAX_PARAMS_JSON_STRING) {
```

Comment:
Hey David, thank you for the pull request. We like this change, but we would prefer it if the rc check was separate and before we use the value of nsize. If Nread fails, it would be better to just fall through. 

```
if (rc == sizeof(nsize)){
    hsize = ntohl(nsize);
    if (hsize > 0 && hsize <= MAX_PARAMS_JSON_STRING) {
        ...
    }
}
```
Also, I realize it's not there now, but adding a print statement else if it fails the size checks would be helpful. Let us know what you think!